File: old/commands/status.py
"""Status command orchestration."""
from dataclasses import dataclass
from cli import PCT
from libs.logger import get_logger
from libs.command import Command
from services.lxc import LXCService
from services.pct import PCTService
logger = get_logger(__name__)


@dataclass
class Status(Command):
    """Status command class."""
    lxc_service: LXCService = None
    pct_service: PCTService = None

    def run(self, args):
        """Show current lab status."""
        # Connect LXC service (injected via DI)
        if not self.lxc_service.connect():
            logger.error("Failed to connect to Proxmox host %s", self.cfg.proxmox_host)
            return
        try:
            logger.info("=" * 50)
            logger.info("Lab Status")
            logger.info("=" * 50)
            # Check containers
            logger.info("Containers:")
            list_cmd = PCT().status()
            result, _ = self.lxc_service.execute(list_cmd)
            if result:
                logger.info(result)
            else:
                logger.info("  No containers found")
            # Check templates
            template_dir = self.cfg.proxmox_template_dir
            logger.info("Templates:")
            template_cmd = f"ls -lh {template_dir}/*.tar.zst 2>/dev/null || echo 'No templates'"
            result, _ = self.lxc_service.execute(template_cmd)
            if result:
                logger.info(result)
            else:
                logger.info("  No templates found")
        finally:
            if self.lxc_service:
                self.lxc_service.disconnect()

File: src/commands/status.go
package commands

import (
	"fmt"
	"enva/cli"
	"enva/libs"
	"enva/services"
)

// Status handles status display
type Status struct {
	cfg        *libs.LabConfig
	lxcService *services.LXCService
	pctService *services.PCTService
}

// NewStatus creates a new Status command
func NewStatus(cfg *libs.LabConfig, lxcService *services.LXCService, pctService *services.PCTService) *Status {
	return &Status{
		cfg:        cfg,
		lxcService: lxcService,
		pctService: pctService,
	}
}

// Run executes the status command
func (s *Status) Run() error {
	logger := libs.GetLogger("status")
	if !s.lxcService.Connect() {
		logger.Error("Failed to connect to Proxmox host %s", s.cfg.LXCHost())
		return fmt.Errorf("failed to connect to Proxmox host")
	}
	defer s.lxcService.Disconnect()
	logger.Info("==================================================")
	logger.Info("Lab Status")
	logger.Info("==================================================")
	logger.Info("Containers:")
	listCmd := cli.NewPCT().Status()
	result, _ := s.lxcService.Execute(listCmd, nil)
	if result != "" {
		logger.Info(result)
	} else {
		logger.Info("  No containers found")
	}
	templateDir := s.cfg.ProxmoxTemplateDir()
	logger.Info("Templates:")
	templateCmd := fmt.Sprintf("ls -lh %s/*.tar.zst 2>/dev/null || echo 'No templates'", templateDir)
	result, _ = s.lxcService.Execute(templateCmd, nil)
	if result != "" {
		logger.Info(result)
	} else {
		logger.Info("  No templates found")
	}
	return nil
}
